src/constraints/calculate_constraint_residual_gradient.hpp

```diff
@@ -22,10 +21,10 @@ struct CalculateConstraintResidualGradient {
     Kokkos::View<size_t* [2][2]>::const_type node_col_range;
     Kokkos::View<double* [3]>::const_type X0_;
     Kokkos::View<double* [3][3]>::const_type axis;
-    View_N::const_type control;
-    View_Nx7::const_type constraint_u;
-    View_Nx7::const_type node_u;
-    View_N Phi_;
+    Kokkos::View<double*>::const_type control;
+    Kokkos::View<double* [7]>::const_type constraint_u;
+    Kokkos::View<double* [7]>::const_type node_u;
+    Kokkos::View<double*> Phi_;
```

Comment:
Correct, the types did not change.  There are a couple things that this type of change does.  

One, it makes how the types are specified more consistent with the other variables in this kernel.  I think doing it this way is better than introducing a bunch of new types into the types.hpp folder (for example, we would need a `View_sizeT_Nx2x2` and a `View_Nx6x12` just looking at this kernel alone.  

Two, it gives us the freedom to adjust the types here as necessary independent of other uses of those aliases.  This could be changing the layout or data types or using template parameters in the kernel to control them.

Three, it let me remove the include of types.hpp from the kernel, so everything we need to know is defined in fewer places

I can see the point of the types.hpp to avoid verbosity in some of our higher-level code, but you need to be well versed in the details of Kokkos to understand any of our kernel code - just look at our `Kokkos::Array` with unmanged View maneuver, for example, or our use of scratch memory and team-based loops.  At this point, global aliases don't really help with readability and can actually hurt when we have to use them side-by-side with View types that don't have aliases.